webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -425,6 +427,46 @@ export function getChannelTimezones(channelId: string): ActionFuncAsync<string[]
     };
 }
 
+export function getUsersCountByStatus(channelId: string, status: string): ActionFuncAsync<number> {
+    return async (dispatch, getState) => {
+        const options = {status};
+        const pageSize = 200;
+
+        let profiles = [];
+
+        let page = 0;
+        for (;;) {
+            let data: UserProfile[];
+
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                data = await Client4.getProfilesInChannel(channelId, page, pageSize, '', options);
+            } catch (error) {
+                forceLogoutIfNecessary(error, dispatch, getState);
+                dispatch(logError(error));
+                return {error};
+            }
+
+            if (!data) {
+                break;
+            }
+
+            profiles.push(...data);
+
+            if (data.length >= pageSize) {
+                ++page;
+            } else {
+                break;
+            }
+        }
+
+        const currentUserId = getCurrentUserId(getState());
+        profiles = profiles.filter((userProfile) => userProfile.id !== currentUserId);
+
+        return {data: profiles.length};
```

Comment:
Nit (feel free to ignore): is there a need to filter here? Assuming the status of the current user is known and easily retrievable (I guess this is true, but don't have the details), we can check against the passed `status` and return either `profiles.length` if it's not the same or `profiles.length-1` if it is the same.